pkg/controller/datavolume-controller.go

```diff
@@ -1122,73 +1171,56 @@ func (r *DatavolumeReconciler) getSnapshotClassForSmartClone(dataVolume *cdiv1.D
 	pvc := &corev1.PersistentVolumeClaim{}
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: sourcePvcNs, Name: sourcePvcSpec.Name}, pvc); err != nil {
 		if k8serrors.IsNotFound(err) {
-			r.log.V(3).Info("Source PVC is missing", "source namespace", sourcePvcSpec.Namespace, "source name", sourcePvcSpec.Name)
+			log.Info("Source PVC is missing", "source namespace", sourcePvcSpec.Namespace, "source name", sourcePvcSpec.Name)
+			return false, nil
```

Comment:
why err is now nil?
previously it stated source PVC not found, and probably the whole operation was retried

Reply:
So before any error would just have us switch to host assisted clone and not retry at all. When I refactored I tried to decide which failures are 'recoverable' aka a retry might fix the problem, and those returned an error. And which failures are permanent (like no CRDs existing) which should switch to host assisted clone.

In this case if err == nil -> host assisted clone
if err != nil -> retry reconcile.

I think I agree a missing source is a retry not a switch to host assisted clone. But I suspect further down it will generate an error because we can't clone

Reply:
So I did some digging here and there is actual code that depends on us not returning an error here, and it will error later. So I am leaving this one nil.